src/app/address/controllers/addressResults.js

```diff
@@ -15,23 +15,19 @@ class AddressResultsController extends BaseController {
         const allAddresses = req.sessionModel.get("searchResults");
 
         const choosenAddress = allAddresses.find(
-          (address) => address.label === selectedAddress
+          (address) => address.text === selectedAddress
         );
```

Comment:
Currently the above find with run at O(n) time. 

I could speed this up by using a map or by setting the pretty print text/label as an object key and the address as the value eg -
`
  { "my address 123" : 
     { 
         "buildingNumber" : 1,
        ...
     }
  }  
`

However I dont think we will be dealing with millions of address so I stuck with a simple array.find.